src/hooks/useScanResults.ts

```diff
@@ -134,9 +138,34 @@ const dataSelectorForScanResults = (data: InfiniteData<TaskRunKind[], unknown>):
   const taskRuns = data?.pages?.flatMap((page) => page) ?? [];
   return processScanTaskRuns(taskRuns);
 };
-export const useKarchScanResults = (pipelineRunName: string): [ScanResults, boolean, unknown] => {
+export const useKarchScanResults = (
+  pipelineRunName: string,
+  pipelineRunCreationTimestamp: string,
+): [ScanResults, boolean, unknown] => {
   const namespace = useNamespace();
 
+  const fieldSelector = React.useMemo(() => {
+    if (!pipelineRunCreationTimestamp) {
+      // eslint-disable-next-line no-console
+      console.error(
+        'useKarchScanResults: pipelineRunCreationTimestamp is required for KubeArchive queries',
+      );
+      return undefined;
+    }
+
+    try {
+      const creationTime = new Date(pipelineRunCreationTimestamp);
+      const timestampAfter = pipelineRunCreationTimestamp;
+      const timestampBefore = new Date(creationTime.getTime() + 24 * 60 * 60 * 1000).toISOString();
+
+      return `creationTimestampAfter=${timestampAfter},creationTimestampBefore=${timestampBefore}`;
```

Comment:
hmm, maybe we could reuse the `convertFilterToKubearchiveSelectors` utility? 🤔 wdyt @sahil143?